src/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -343,9 +343,7 @@ public void onClick(View v) {
     @Override
     public void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
-        // check if its a oauth SALS callback intent
-        if(intent.getAction().equals(ACTION_VIEW) &&
-                intent.getData().getHost().equals("android.owncloud.com")) {
+        if(intent.getAction().equals(ACTION_VIEW)) {

```

Comment:
Don't you need `intent.getData().getHost().equals("android.owncloud.com"))` finally?

Reply:
This statement should already filter the allowed statement:
https://github.com/owncloud/android/blob/96c141cc899f9c4e689a81618982be7284885567/AndroidManifest.xml#L203